tree/ntuple/v7/test/ntuple_compat.cxx

```diff
@@ -58,3 +64,83 @@ TEST(RNTupleCompat, FeatureFlag)
       EXPECT_THAT(err.what(), testing::HasSubstr("unsupported format feature: 137"));
    }
 }
+
+TEST(RNTupleCompat, FwdCompat_FutureNTuple)
+{
+   using ROOT::Experimental::RXTuple;
+
+   constexpr static const char *kNtupleObjName = "ntpl";
+
+   FileRaii fileGuard("test_ntuple_compat_fwd_compat_future_ntuple.root");
+
+   // Write an RXTuple to disk. It is a simulacrum of a future version of RNTuple, with additional fields and a higher
+   // class version.
+   {
+      auto file = std::unique_ptr<TFile>(
+         TFile::Open(fileGuard.GetPath().c_str(), "RECREATE", "", ROOT::RCompressionSetting::ELevel::kUncompressed));
+      auto xtuple = RXTuple{};
+      file->WriteObject(&xtuple, kNtupleObjName);
```

Comment:
We later assume/rely-on the fact that the file is small.  If something goes wrong (either via some corruption or a bug introduced down the line), we could add here (`file->GetEND()`) a check on the file size being small to ease debugging of the test failure.